components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/Utils.java

```diff
@@ -773,6 +773,19 @@ public static boolean isGraphQLSubscriptionRequest(MessageContext messageContext
                         && (boolean) messageContext.getProperty(APIConstants.GRAPHQL_SUBSCRIPTION_REQUEST));
     }
 
+    /**
+     * Evaluate current request transport and message context to check if it is a MCP request execution path.
+     *
+     * @param messageContext MessageContext
+     * @return true if MCP request execution path
+     */
+    public static boolean isMCPRequest(MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                getAxis2MessageContext();
+        String apiType = (String) messageContext.getProperty(APIMgtGatewayConstants.API_TYPE);
+        return apiType != null && apiType.equals(APIConstants.API_TYPE_MCP);
```

Comment:
We can write this as APIConstants.API_TYPE_MCP.equals(apiType) to avoid the null check.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/13401/commits/d13ffe034a2658724cf3726ba2a77d87ff0a48b7